pdns/syncres.cc

```diff
@@ -945,6 +949,37 @@ int SyncRes::doResolveNoQNameMinimization(const DNSName &qname, const QType &qty
 
       return res;
     }
+
+    /* if we have not found a cached DS (or denial of), now is the time to look for a CNAME */
```

Comment:
Don't like the repeated block a lot but I do not see an easy way to get rid of it.

Reply:
Agreed, I tried a few ideas but they all turned out worse than duplicating the block, so I left it that way.